### ISSUE TITLE
Train VOC dataset with lots of 'nan'

### ISSUE THREAD

**Issue by HanYuanyuaner** [AlexeyAB/darknet#884]
![screenshot from 2018-05-23 19-58-40](https://user-images.githubusercontent.com/37731738/40423122-1a88060e-5ec4-11e8-9dd4-7dcb9f187bea.png)
Here is the information, lots of nan.
The parameters in .cfg
![screenshot from 2018-05-23 20-02-38](https://user-images.githubusercontent.com/37731738/40423182-429595d0-5ec4-11e8-8f7b-5862a9d137d5.png)
I have changed batch and subdivisions.
Can any one tell me why so many 'nan'
@AlexeyAB

**Comment by HanYuanyuaner** [comment]
At beginning, the avg like this
![screenshot from 2018-05-24 13-45-31](https://user-images.githubusercontent.com/37731738/40466596-f5cea33c-5f58-11e8-81e7-9f294f0a3437.png)
after running some time it become like this
![screenshot from 2018-05-24 13-46-00](https://user-images.githubusercontent.com/37731738/40466618-0c390f54-5f59-11e8-9ef0-c5bd150d5787.png)
the avg become 'nan'
@AlexeyAB 
Do I need to change the learning rate?

**Comment by AlexeyAB** [comment]
@HanYuanyuaner 

Training goes wrong.

* What params are in the Makefile?
* What batch= and subdivisions= in cfg?
* Did you check your dataset by using Yolo_mark?

**Comment by HanYuanyuaner** [comment]
1.My parameters in Makefile
![screenshot from 2018-05-24 17-45-12](https://user-images.githubusercontent.com/37731738/40477938-5039cd76-5f7a-11e8-85db-c199b775c7d4.png)
my GPU is Tesla p100 16G
2.batch = 64 subdivisions = 16
3. I use VOC dataset, 
![screenshot from 2018-05-24 17-48-28](https://user-images.githubusercontent.com/37731738/40478080-b1d1014e-5f7a-11e8-82b7-7e408057fa85.png)
the structure of image 
![screenshot from 2018-05-24 17-49-13](https://user-images.githubusercontent.com/37731738/40478113-c9e20bac-5f7a-11e8-82b8-543d543d6939.png)

**Comment by AlexeyAB** [comment]
Try to use `CUDNN_HALF=0`

`CUDNN_HALF=1` is for GPU Volta (Tesla V100, Titan V, DGX2, ...)

**Comment by HanYuanyuaner** [comment]
@AlexeyAB 
I followed your advise and re-trained the model successfully. 
Thank you so much.
Another question is that can I calculate the mAP of my re-trained model?

**Comment by AlexeyAB** [comment]
@HanYuanyuaner Yes, you can calculate mAP for your own model. Just set valid= param in your `obj.data` file to your validation or test dataset.
Or just set `valid=train.txt` in `obj.data`-file.
